webserver/src/main/java/io/kestra/webserver/controllers/NamespaceFileController.java

```diff
@@ -109,6 +116,23 @@ public void delete(
         storageInterface.delete(tenantService.resolveTenant(), toNamespacedStorageUri(namespace, path));
     }
 
+    @ExecuteOn(TaskExecutors.IO)
+    @Get(uri = "files/distinct-namespaces", produces = MediaType.TEXT_JSON)
+    @Operation(tags = {"Files"}, summary = "List existing namespace folders")
+    public List<String> distinctNamespaces() throws IOException, URISyntaxException {
+        List<String> storageNamespaces =  distinctNamespacesFiles();
+        List<String> flowNamespaces = flowRepository.findDistinctNamespace(tenantService.resolveTenant());
+
+        return Stream.concat(storageNamespaces.stream(), flowNamespaces.stream()).distinct().toList();
+    }
+
+    protected List<String> distinctNamespacesFiles() throws IOException, URISyntaxException {
+        return storageInterface.list(tenantService.resolveTenant(), new URI(""))
```

Comment:
It doesn't work with S3, idk for GCS :thinking: 
![image](https://github.com/kestra-io/kestra/assets/37618489/14a48ebf-0bfe-4085-9e7f-38665e2c422e)
![image](https://github.com/kestra-io/kestra/assets/37618489/49c13d23-0121-4423-9de4-6f3b631f472f)

Btw I think we have an issue as the extension is creating namespace file without a leading '/' and it seems to have an impact on some storage (s3 does, idk for gcs). We might need to discuss whether it is a problem and if we should keep the forward '/' also for namespace files (currently it is only the case for executions files)
![image](https://github.com/kestra-io/kestra/assets/37618489/ab889fc4-0c05-46c9-830f-5ed1bf2640a0)
This change should be made to fix this but we might need to discuss about it
![image](https://github.com/kestra-io/kestra/assets/37618489/2d3d332d-740e-4c96-bfb8-17eb21de6d89)

Also, idk if that's the case for localstorage but for eg. in S3 you can't list "" & "/", not sure if it's an implementation issue or a SDK limitation, that's an issue we should adress quickly

Reply:
We should never list all the internal storage like that.
What you want if I understood, is to list all namespaces that have files. But this is not what you did here as namespaces can have execution data but not files, and in this case, you will still include it in the list.

Moreover, listing like this will bypass security checks on EE so it's really not a good idea.